src/components/Search/index.tsx

```diff
@@ -196,7 +196,13 @@ function Search({queryJSON, onSearchListScroll, contentContainerStyle}: SearchPr
     const shouldShowLoadingMoreItems = !shouldShowLoadingState && searchResults?.search?.isLoading && searchResults?.search?.offset > 0;
     const isSearchResultsEmpty = !searchResults?.data || SearchUIUtils.isSearchResultsEmpty(searchResults);
     const prevIsSearchResultEmpty = usePrevious(isSearchResultsEmpty);
-    const data = searchResults === undefined ? [] : SearchUIUtils.getSections(type, status, searchResults.data, searchResults.search);
```

Comment:
FYI: `data` was used as a dependency in a useEffect below, which caused the infinity loop (this component is not compiled because we ignore React rules 4 times, normally it would be taken care of)

Reply:
Thanks for the fix. 
Two things come to mind:
 - how come this did not surface previously?
 - I just remembered that there is a `CONST.EMPTY_ARRAY` for similar cases 😅 but `useMemo` feels better

Reply:
Those are great questions. I was asking myself the same thing since we didn't really change any behavior besides changing the react compiler.

Reply:
The only explanation I came up with is that `Search` component was somehow compiled before updating the library 🤔

Reply:
hey @blazejkustra @luacmartins I finally found some time to investigate this and try to understand why did broke.
The theory that Błażej posted above is 100% correct.

I have checked out on a commit before the introduction of the updated (bumped?) react-compiler and run healthcheck then. The component `Search/index.ts` was previously compiled without any problems. And I think this compilation and memoization is what prevented the bug.
Then I checked out onto a commit that was before this fix but after updating react-compiler - and Search is no longer compiled. So this is the reason in the past there was no infinite effect but now there was.

Finally you can test yourself that Search is also not compiled on newest main. Here is the reason:
```
Failed to compile src/components/Search/index.tsx:182:78. Reason: Ref values (the `current` property) may not be accessed during render. (https://react.dev/reference/react/useRef)
```